src/mork/edx/database.py

```diff
@@ -0,0 +1,77 @@
+"""Mork database connection."""
+
+from datetime import datetime, timedelta
+from typing import Optional
+
+from sqlalchemy import URL, create_engine, select
+from sqlalchemy.orm import Session, load_only
+
+from mork.conf import settings
+from mork.edx.models import User
+
+
+class OpenEdxDB:
+    """Class to connect to the Open edX database and create models."""
+
+    session = None
+    User = None
+
+    def __init__(self, engine=None, session=None):
+        """Initialize SqlAlchemy engine and session."""
+        if engine is not None:
+            self.engine = engine
+        else:
+            url = URL.create(
+                drivername=settings.EDX_DB_ENGINE,
+                username=settings.EDX_DB_USER,
+                password=settings.EDX_DB_PASSWORD,
+                host=settings.EDX_DB_HOST,
+                port=settings.EDX_DB_PORT,
+                database=settings.EDX_DB_NAME,
+            )
+            self.engine = create_engine(
+                url, echo=settings.EDX_DB_DEBUG, pool_pre_ping=True
+            )
+        if session is not None:
+            self.session = session
+        else:
+            self.session = Session(self.engine)
+
+        self.User = User
+
+    def get_inactive_users(
```

Comment:
I didn't find anything in the SQLAlchemy documentation about where should be declared this kind of method. I don't think doing it here is a good idea. The joanie import project was more a one shot project and we will delete it soon.
Here it is not the case and I think this kind of method should not be all declare in this class.

In an ORM like doctrine, a repository can be used and it contains all this kind of method related to a model

Reply:
You're right, it should not be in this class. As this particular query will be used in the two tasks (warning the users, delete the users), I think it should be nice to have a function for that.
What about having it in a `utils.py`?

Reply:
I pushed a proposal where I add a class method to the User model through a mixin. Let me know what you think of it!